### PR TITLE
[Script] Script Module

### PR DESCRIPTION
**Pull request by hgt312**
A re-submission of https://github.com/meta-project/torch_mnm/pull/124

cc @awslabs/raf-reviewer

### CODE REVIEW

File: ratex/csrc/pass_ext/wrap_identity.cpp [hunk 1]
```diff
@@ -34,18 +34,34 @@ namespace wrap_identity {
...
 using namespace raf::op;
 
+// This pass replaces the output tuple: %ret = (%x_0, %x1, ..., %x_n)
+// 1. constants are copied: if %x_i is a constant node, it is replaced by
+// let %new_x_i = mnm.op.copy(%x_i)
+// 2. input parameters are copied: if %x_i is a input parameter %p_j, it is replaced by
+// let %new_x_i = mnm.op.copy(%p_j)
+// 3. if outputs are duplicate: %ret = (%x0, %x1, ..., %xi, ...%xi, ...)
+// let %new_x_i = mnm.op.copy(%x_i)
+// let %ret = (%x0, %x1, ..., %xi, ..., %new_x_i, ...)
```
Comment (comaniac) [inline ratex/csrc/pass_ext/wrap_identity.cpp:44]:
```suggestion
/*! \brief This pass processes the output tuple to make it RAF friendly.
 * Given an output tuple %ret = (%x_0, %x1, ..., %x_n), 
 * 1. constants are copied. If %x_i is a constant node, it becomes
 *     let %new_x_i = raf.op.copy(%x_i)
 * 2. input parameters are copied. If %x_i is a input parameter %p_j, it becomes
 *     let %new_x_i = raf.op.copy(%p_j)
 * 3. duplicated fields are copied. If %xi is outputed twice, it becomes
 *     let %new_x_i = raf.op.copy(%x_i)
 *     let %ret = (%x0, %x1, ..., %xi, ..., %new_x_i, ...)
 */
```

File: ratex/jit/script.py [hunk 2]
```diff
@@ -242,56 +240,64 @@ def script(module: torch.nn.Module):
...
-
-        positional_args = get_positional_args(param_names, *args, **params)
-        return RelayFunction.apply(func, inplace_update_map, *positional_args)
-
-    return wrapper
+    cloned_module = copy.deepcopy(module)
+    cloned_module = cloned_module.cpu()
+
+    class ScriptModule(torch.nn.Module):
+        """A wrapper of module to run on Meta."""
```
Comment (comaniac) [inline ratex/jit/script.py:247]:
```suggestion
        """A wrapper of module to run on RAF."""
```

File: ratex/jit/script.py [hunk 3]
```diff
@@ -242,56 +240,64 @@ def script(module: torch.nn.Module):
...
+            # TODO: use torch.jit.script
+            assert len(args) == 1, f"Only support single input for now, but got {len(args)}"
+            assert not kwargs, "Do not support kwargs yet"
+            shape_n_dtype = (list(args[0].shape), str(args[0].dtype).rsplit(".", maxsplit=1)[-1])
+            cache_key = (hash_torch_module(cloned_module), str(shape_n_dtype))
+            if cache_key in JIT_CACHE:
+                # Cache hit.
+                func, param_names, inplace_update_map = JIT_CACHE[cache_key]
+            else:
+                # Cache miss. Generate a Meta function and apply a series of transformations.
```
Comment (comaniac) [inline ratex/jit/script.py:273]:
```suggestion
                # Cache miss. Generate a RAF function and apply a series of transformations.
```

File: scripts/build_ratex.sh [hunk 4]
```diff
@@ -10,16 +10,27 @@ export CC=gcc
...
   exit 1
 fi
 
-echo "Building ratex wheels..."
-python3 -m pip install glob2 filelock
-rm -rf ./build/pip/public/ratex
-python3 setup.py bdist_wheel -d ./build/pip/public/ratex
-pip3 install ./build/pip/public/ratex/*.whl --upgrade --force-reinstall --no-deps
+if [ $DEVELOP -eq 0 ]; then
+  echo "Building razor wheels..."
```
Comment (comaniac) [inline scripts/build_ratex.sh:26]:
```suggestion
  echo "Building ratex wheels..."
```

File: scripts/build_ratex.sh [hunk 5]
```diff
@@ -10,16 +10,27 @@ export CC=gcc
...
+  rm -rf ./build/pip/public/razor
+  python3 setup.py bdist_wheel -d ./build/pip/public/razor
+  pip3 install ./build/pip/public/razor/*.whl --upgrade --force-reinstall --no-deps
+else
+  echo "Building razor develop..."
```
Comment (comaniac) [inline scripts/build_ratex.sh:31]:
```suggestion
  echo "Building ratex develop..."
```

**Review by comaniac — COMMENTED** [review]
Mostly just docstring and comment issues.
Also, do we need a unit test?

cc @hzfan

**Review by comaniac — APPROVED** [review]
LGTM.

**Comment by comaniac** [comment]
@hzfan @zachzzc please also review this PR and feel free to merge if no comments. Thanks.

**Review by hzfan — APPROVED** [review]
LGTM.

**Comment by hzfan** [comment]
Thanks @hgt312 @comaniac

**Pull request merged**